htlcswitch/link.go

```diff
@@ -870,6 +874,10 @@ func (l *channelLink) htlcManager() {
 
 	l.log.Infof("HTLC manager started, bandwidth=%v", l.Bandwidth())
 
+	// Notify any clients that the link is now in the switch via an
+	// ActiveLinkEvent.
+	l.cfg.NotifyActiveLink(*l.ChannelPoint())
```

Comment:
Perhaps this should be done _after_ we've processed the channel reest messages? Or does this get in the way fo the new logic ot handle these messages as if they were any other message?

Reply:
We can't do this - we wait for this notification in the peer *before* sending any messages (including channel reestablish).